v3/src/hooks/use-adjust-header-overflow.ts

```diff
@@ -0,0 +1,107 @@
+import { useEffect, useRef, useState } from 'react'
+import { measureText } from './use-measure-text'
+
+const kPaddingBuffer = 5 // Button width is 5px smaller because of parent padding
+
+// Hook to split headers into 2 rows and ellide the 2nd line if it doesn't fit
```

Comment:
```suggestion
// Hook to split headers into 2 rows and elide the 2nd line if it doesn't fit
```
"Ellipsis" has two L's but "elide" only has one. 🤷